### ISSUE TITLE
output_random_32_1000.txt

### ISSUE THREAD

**Issue by JakubKana** [beaunus/stanford-algs#59]
stanford-algs/testCases/course3/assignment3HuffmanAndMWIS/question1And2/output_random_32_1000.txt

I belive that right answer is 19 and 9 all others are correct in my huffman algorithm.

stanford-algs/testCases/course3/assignment3HuffmanAndMWIS/question1And2/output_random_48_10000.txt

30 and 12

**Comment by dgrcode** [comment]
Hi!

I just ran the code I wrote for the actual assignment and it passed all the tests, included the two you mentioned.

For `random_32_1000` the output in the repo is `[16, 9]` and your output is `[19, 9]`.
For `random_48_10000` the output in the repo is `[26, 12]` and your output is `[30, 12]`.

In both cases the min matches but the max doesn't. The fact that the rest of the tests pass with your algorithm makes me think the counting is done correctly, so I'm guessing the issue is related with the logic of the algorithm.

The fact that in both cases your max output is bigger would also make sense if there was a logic issue with your implementation.

I'd recommend trying to look for possible edge cases that your algorithm wouldn't support at the moment. Have a look at the slides, one file named `algo2-greedy-huffman4-typed.pdf`. There they talk about "Huffman’s Algorithm: A More Complex Example". In that case the input makes the tree grow in a way that is slightly different than most of the time, and that way of growing makes the tree be smaller, and therefore the max would be smaller too.

How are you keeping track of the node that should be merged next?

That's just a guess and it could be something else, but it's definitely worth checking.

Let me know how it goes!
